src/main/java/Duke.java

```diff
@@ -1,10 +1,103 @@
+import java.io.File;
+import java.time.LocalDate;
+
+/**
+ * This is a simulation of a chat bot called Duke.
+ */
 public class Duke {
+    private TaskList tasks;
+    private Storage storage;
+    private Ui ui;
+
+    /**
+     * Constructs a new Duke with the given data storage path.
+     *
+     * @param path The file path of Duke data file.
+     */
+    public Duke(String path) {
+        this.ui = new Ui();
+        this.storage = new Storage(path);
+        this.tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Runs Duke to take in user input and display reply until termination.
+     * Task data is stored when running.
+     */
+    public void run() {
+        ui.printLogo();
+        ui.printGreet();
+
+        String input = null;
+        while (true) {
+            input = ui.getInput();
+
+            if (input.equals("bye")) {
+                ui.printExit();
+                break;
+            }
+
+            if (input.equals("list")) {
```

Comment:
It would be better to abstract the commands away into a class called Command with the different subclasses of commands, and add a class called Parser to parse the commands.